src/ducc0/infra/threading.cc

```diff
@@ -537,14 +573,20 @@ void Distribution::thread_map(std::function<void(Scheduler &)> f)
   latch counter(nthreads_);
   std::exception_ptr ex;
   std::mutex ex_mut;
+  // we "copy" the currently active thread pool to all executing threads
+  // during the time of execution and pop it again at the end.
+  // Not sure if this is necessary/useful though.
```

Comment:
Do you plan on handling nested parallelism? Without explicit support it's just a recipe for over-subscription.

Reply:
I don't really expect nested parallelism, but I don't see any reason to exclude it categorically. And if it can occur, then it probably should occur within the same thread pool by default.

Reply:
I could give the workers a "serializing" thread pool which runs any nested parallel regions in a serial fashion on itself. How about that?

Reply:
I would probably leave as is and let the threadpool itself decide how to handle nested parallelism.  Your default threadpool already does this by checking `in_parallel_region`.

In TF, we often send large jobs to the threadpool and allow those jobs to split themselves into subtasks.